protocol/x/revshare/keeper/revshare_test.go

```diff
@@ -268,26 +267,52 @@ func TestValidateRevShareSafety(t *testing.T) {
 }
 
 func TestKeeper_GetAllRevShares_Valid(t *testing.T) {
+	marketId := uint32(1)
 	tests := []struct {
-		name                                  string
-		expectedRevShares                     int
-		monthlyRollingTakerVolumeQuantums     uint64
-		expectedTakerFeeRevSharesPpmByAddress map[string]uint32
-		expectedNetFeeRevSharesPpmByAddress   map[string]uint32
-		setup                                 func(tApp *testapp.TestApp, ctx sdk.Context,
+		name              string
+		fill              clobtypes.FillForProcess
+		expectedRevShares []types.RevShare
+		setup             func(tApp *testapp.TestApp, ctx sdk.Context,
 			keeper *keeper.Keeper, affiliatesKeeper *affiliateskeeper.Keeper)
 	}{
 		{
-			name:                              "Valid revenue share from affiliates, unconditional and market mapper",
-			expectedRevShares:                 4,
-			monthlyRollingTakerVolumeQuantums: 1_000_000_000_000, // 1 million USDC
-			expectedTakerFeeRevSharesPpmByAddress: map[string]uint32{
-				constants.BobAccAddress.String(): 150_000, // 15% of taker fees
-			},
-			expectedNetFeeRevSharesPpmByAddress: map[string]uint32{
-				constants.AliceAccAddress.String(): 400_000, // 40% of net fees
-				constants.BobAccAddress.String():   200_000, // 20% of net fees
+			name: "Valid revenue share from affiliates, unconditional and market mapper",
+			expectedRevShares: []types.RevShare{
+
+				{
+					Recipient:         constants.BobAccAddress.String(),
+					RevShareFeeSource: types.REV_SHARE_FEE_SOURCE_TAKER_FEE,
+					RevShareType:      types.REV_SHARE_TYPE_AFFILIATE,
+					QuoteQuantums:     big.NewInt(1_500_000),
+				},
+				{
+					Recipient:         constants.BobAccAddress.String(),
+					RevShareFeeSource: types.REV_SHARE_FEE_SOURCE_NET_FEE,
+					RevShareType:      types.REV_SHARE_TYPE_UNCONDITIONAL,
+					QuoteQuantums:     big.NewInt(2_400_000),
+				},
+				{
+					Recipient:         constants.AliceAccAddress.String(),
+					RevShareFeeSource: types.REV_SHARE_FEE_SOURCE_NET_FEE,
+					RevShareType:      types.REV_SHARE_TYPE_UNCONDITIONAL,
+					QuoteQuantums:     big.NewInt(3_600_000),
+				},
+				{
+					Recipient:         constants.AliceAccAddress.String(),
+					RevShareFeeSource: types.REV_SHARE_FEE_SOURCE_NET_FEE,
+					RevShareType:      types.REV_SHARE_TYPE_MARKET_MAPPER,
+					QuoteQuantums:     big.NewInt(1_200_000),
+				},
 			},
+			fill: clobtypes.CreatePerpetualFillForProcess(
```

Comment:
This function actually makes the code less readable, could we just replace this with a struct literal whereever it's used? For example it's hard for reader to tell what these fields are

Reply:
If we want to get rid of the function and update the struct directly, we have to rename the fields to be capitalized(same as the interface - causing errors) 
two options here:
1. Remove interface entirely and stick with structs
2. Change the anme of underlying stuct vars(like PerpetualMarketAddr or MarketAddr_) which might be a little more confusing

Reply:
im leaning 1